src/coffee/sync/utils/product.coffee

```diff
@@ -100,19 +100,27 @@ class ProductUtils extends BaseUtils
       actions.push action if action
     actions
 
-  matchesByIdOrKeyOrSku: (variant1, variant2) ->
-    variant1 and variant2 and (
-      (!isNil(variant1.id) and variant1.id == variant2.id) or
-      (!isNil(variant1.key) and variant1.key == variant2.key) or
-      (!isNil(variant1.sku) and variant1.sku == variant2.sku)
-    )
+  matchesBySkuOrKeyOrId: (variant1, variant2) ->
+    @matchesBySku(variant1, variant2) or @matchesByKey(variant1, variant2) or @matchesById(variant1, variant2)
 
+  matchesById: (variant1, variant2) ->
+    variant1 and variant2 and not isNil(variant1.id) and variant1.id == variant2.id
+
+  matchesByKey: (variant1, variant2) ->
+    variant1 and variant2 and not isNil(variant1.key) and variant1.key == variant2.key
+
+  matchesBySku: (variant1, variant2) ->
+    variant1 and variant2 and not isNil(variant1.sku) and variant1.sku == variant2.sku
+
+  # match variant against variants in list - match first by sku, then by key and then by id
   findVariantInList: (variant, variantList) ->
-    variantList.find (testedVariant) =>
-      @matchesByIdOrKeyOrSku(testedVariant, variant)
+    return variantList.find((oldVariant) => @matchesBySku(variant, oldVariant)) or
+      variantList.find((oldVariant) => @matchesByKey(variant, oldVariant)) or
+      variantList.find((oldVariant) => @matchesById(variant, oldVariant)) or
+      undefined # if not found, return undefined
 
   buildChangeMasterVariantAction: (newMasterVariant, oldMasterVariant) ->
-    if newMasterVariant and oldMasterVariant and not @matchesByIdOrKeyOrSku(newMasterVariant, oldMasterVariant)
+    if newMasterVariant and oldMasterVariant and not @matchesBySkuOrKeyOrId(newMasterVariant, oldMasterVariant)
```

Comment:
Should not we change the order also here and [here](https://github.com/sphereio/sphere-node-sdk/pull/273/files#diff-a94247bc9925c68272e1e3c1213c3556L119) and [here](https://github.com/sphereio/sphere-node-sdk/pull/273/files#diff-a94247bc9925c68272e1e3c1213c3556R145)? I think it might be better to always use sku if defined and fallback on internal only when nothing else possible.

Reply:
Hi @butenkor .. updated here c8afbe7